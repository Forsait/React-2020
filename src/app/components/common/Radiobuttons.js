// @flow
import React, { useEffect, useRef } from 'react';

// $FlowFixMe
import { createUseStyles } from 'react-jss';

const useStyles = createUseStyles({
  radio_container: {
    display: 'flex',
    alignItems: 'center',
    textTransform: 'uppercase',
    opacity: 0.8,
    color: '#FFFFFF',
    fontSize: '16px',
    fontWeight: 500,
    letterSpacing: '0.89px',
    lineHeight: '19px',
  },

  radio_list: {
    marginLeft: '20px',
  },

  radio_item: {
    position: 'relative',
    fontSize: '15px',

    '&:first-child': {
      '& span': {
        borderRadius: '5px 0 0 5px',
      },
    },

    '&:last-child': {
      '& span': {
        borderRadius: '0 5px 5px 0',
      },
    },

    '& span': {
      display: 'inline-block',
      padding: '8px 20px',
      backgroundColor: 'rgba(66,66,66,0.68)',
      cursor: 'pointer',
    },

    '& input': {
      position: 'absolute',
      width: 0,
      height: 0,
      opacity: 0,
    },

    '$ input:checked': {
      '& ~span': {
        backgroundColor: '#f65261',
      },

    },
  },

  title: {
    opacity: 0.8,
    fontSize: '15px',
    letterSpacing: 0,
    lineHeight: '19px',
  },
});

interface Radio {
  key: string,
  title: string,
}

interface Props {
  className: string,
  title: string,
  list: Radio[],
  change: (e: any) => void,
}

export default function Radiobuttons(props: Props) {
  const classes = useStyles();
  const innerRef = useRef(null);

  const changeRadio = (e: any) => {
    props.change(e.target.value);
  };

  useEffect(() => {
    const div = innerRef.current;
    // $FlowFixMe
    div.addEventListener('change', changeRadio);
    return () => {
      // $FlowFixMe
      div.removeEventListener('change', changeRadio);
    };
  }, []);

  const { list, title, className } = props;

  const arr = list.map((item, i) => {
    const id = title + i;
    return (
      /* eslint-disable */
      <label key={id} className={classes.radio_item}>
        <input type="radio" name={title} value={item.key} id={id} defaultChecked={i === 0} />
        <span>{item.title}</span>
      </label>
    );
  });

  return (
    <div className={`${classes.radio_container} r-radio ${className}`}>
      <span className={classes.title}>{title}</span>
      {/* $FlowFixMe */}
      <div className={classes.radio_list} ref={innerRef}>
        {arr}
      </div>
    </div>
  );
}
