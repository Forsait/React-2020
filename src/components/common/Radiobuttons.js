import React from 'react';

import styles from './Radiobuttons.module.scss';

export default class Radiobuttons extends React.Component {

  constructor(props) {
    super(props);
    this.changeRadio = this.changeRadio.bind(this);
  }

  changeRadio(e) {
    this.props.change(e.target.value);
  }

  render() {
    const arr = this.props.list.map((item, i) => {
      const id = this.props.title + i;
      return <label key={id} className={styles.radio_item}>
        <input type="radio" name={this.props.title} value={item.key} id={id} defaultChecked={i === 0}/>
        <span>{item.title}</span>
      </label>
    })

    return (
      <div className={styles.radio_container + ' ' + this.props.className}>
        <span className={styles.title}>{this.props.title}</span>
        <div className={styles.radio_list} onChange={this.changeRadio}>
          {arr}
        </div>
      </div>
    )
  }
}