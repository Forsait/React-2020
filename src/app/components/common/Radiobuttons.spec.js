import React from 'react';
import { shallow } from 'enzyme';

import Radiobuttons from './Radiobuttons';

const options = {
  title: 'Title',
  list: [
    {
      key: 'key',
      title: 'title'
    },
    {
      key: 'key 1',
      title: 'title 1'
    }
  ]
};

describe('Radiobuttons component', () => {

  it('should be render correctly', () => {
    const com = shallow(<Radiobuttons title={options.title} list={options.list} />);
    expect(com).toMatchSnapshot()
  });

  it('renders list radiobuttons', () => {
    const com = shallow(<Radiobuttons title={options.title} list={options.list} />);
    expect(com.find('label').length).toEqual(options.list.length);
  });

});