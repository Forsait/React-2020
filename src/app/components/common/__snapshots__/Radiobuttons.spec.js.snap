// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radiobuttons component should be render correctly 1`] = `
<div
  className="undefined r-radio undefined"
>
  <span>
    Title
  </span>
  <div
    onChange={[Function]}
  >
    <label
      key="Title0"
    >
      <input
        defaultChecked={true}
        id="Title0"
        name="Title"
        type="radio"
        value="key"
      />
      <span>
        title
      </span>
    </label>
    <label
      key="Title1"
    >
      <input
        defaultChecked={false}
        id="Title1"
        name="Title"
        type="radio"
        value="key 1"
      />
      <span>
        title 1
      </span>
    </label>
  </div>
</div>
`;
